sklearn/mixture/tests/test_gaussian_mixture.py

```diff
@@ -1040,3 +1040,35 @@ def test_init():
                                max_iter=1, random_state=random_state).fit(X)
 
         assert gmm2.lower_bound_ >= gmm1.lower_bound_
+
+
+def test_gaussian_mixture_setting_best_params():
+    # test that best_params is set appropriately
+    # even if convergence was not reached
+    # also a regression test for issue #18216
+    rnd = np.random.RandomState(0)
+    n_samples = 30
+    X = rnd.uniform(size=(n_samples, 3))
+
+    # specify initialization parameters that won't lead to a convergence
+    means_init = np.array([
+            [0.670637869618158, 0.21038256107384043, 0.12892629765485303],
+            [0.09394051075844147, 0.5759464955561779, 0.929296197576212],
+            [0.5033230372781258, 0.9569852381759425, 0.08654043447295741],
+            [0.18578301420435747, 0.5531158970919143, 0.19388943970532435],
+            [0.4548589928173794, 0.35182513658825276, 0.568146063202464],
+            [0.609279894978321, 0.7929063819678847, 0.9620097270828052],
+    ])
+    precisions_init = np.array([999999.999604483, 999999.9990869573,
+                                553.7603944542167, 204.78596008931834,
+                                15.867423501783637, 85.4595728389735])
+    weights_init = [0.03333333333333341, 0.03333333333333341,
+                    0.06666666666666674, 0.06666666666666674,
+                    0.7000000000000001, 0.10000000000000007]
+
+    gmm = GaussianMixture(covariance_type="spherical", reg_covar=0,
+                          means_init=means_init, weights_init=weights_init,
+                          random_state=rnd, n_components=len(weights_init),
+                          precisions_init=precisions_init)
+    # ensure that no error is thrown during fit
+    gmm.fit(X)
```

Comment:
Could you also please test that the best parameters (`gmm.weights_`, `gmm.means_`, `gmm.covariances_`,
`gmm.precisions_cholesky_`), `gmm.n_iter_` and `gmm.lower_bound_` are also set here? 🙂

Reply:
Thank you for your suggestions! I updated the new test but haven't found a set of params that led to divergence for `BayesianMixture`.